subsys_read_user_input.py

```diff
@@ -36,96 +38,114 @@ class ModeStatus:
 
 class ReadUserInput:
     """
-    Handles the keyboard inputs to manually control the Tello
-    The controls are:
-        - T: Takeoff
-        - L: Land
-        - Space: Emergency 
-        - Arrow keys: Forward, backward, left and right.
-        - Q and D: Counterclockwise and clockwise rotations (yaw)
-        - Z and S: Up and down
-        - Esc: Quit
+    Handles the user inputs to manually control the Tello
+    The controls are defined in the subsys_gamepad.py -> Gamepad class
     """
+
+    joysticks: List[pygame.joystick.Joystick] = []
+    joystick_maps: List[dict] = []
+
     @classmethod
     def setup(cls):
         RunStatus.value = RUN.START
         ModeStatus.value = -1
+        pygame.joystick.init()
+        cls.joysticks = [pygame.joystick.Joystick(j) for j in range(pygame.joystick.get_count())]
+        print(len(cls.joysticks), 'joystick(s) found')
+        for joystick in cls.joysticks:
+            name = joystick.get_name()
+            try:
+                gp_map = [gp['name'] for gp in Gamepad.map_list].index(name)
+                cls.joystick_maps.append(Gamepad.map_list[gp_map])
+                print('Gamepad map successfully loaded for %s', name)
+            except ValueError:
+                print('No configured map for connected gamepad -> using default configuration')
+                gp_map = [gp['name'] for gp in Gamepad.map_list].index('Default')
+                cls.joystick_maps.append(Gamepad.map_list[gp_map])
+            joystick.init()
```

Comment:
What happen when multiple joysticks sends contradictory inputs ?

Reply:
Both inputs are pulled into the pygame event queue. The last event received is then the only input that matters because it overwrites the RCStatus class.
Not the best way of doing I assume, but I feel like having a priority system does not make much sense since the pygame.Joystick index is kind of arbitrary.

Reply:
That is also the behavior i understood from the code. We can merge sicne it won't cause us trouble but this can create further problems : 
If someone decides to implement analogical component (which we might want to do), depending on the quality and the coding of joysticks, some might send "approximate zeros" even when at complete rest which could take over our capacity to control the drone with another joystick